File: src/domain/model/relationship.ts
import {Entity} from './shared.model';

type RelationshipType = '';

export interface Relationship extends Entity {

  characterId1: string;

  characterId2: string;

  type: RelationshipType;

  bookId: string;
}

export const equalRelationships = (relationship1: Relationship, relationship2: Relationship): boolean => relationship1.id === relationship2.id;

File: src/infrastructure/python/model/relationship.py
class Relationship:
    def __init__(self, characterId1, characterId2):
        self.id = name
        self.characterId1 = characterId1
        self.characterId2 = characterId2
        self.type = "undefined"
        self.bookId = "scadrial"

    def __str__(self):
      return self.characterId1 + " <-> " + self.characterId2

    def __hash__(self):
      return hash(self.id)

    def __eq__(self, relationship):
        return relationship.id == self.id
